docs/proposals/object-search.md

```diff
@@ -0,0 +1,160 @@
+    - [Downstream Controller](#downstream-controller)
+    - [Upstream Controller](#upstream-controller)
+    - [Details of api between GM(cloud) and LC(edge)](#details-of-api-between-gmcloud-and-lcedge)
+    - [Flow of object search service creation](#flow-of-object-search-service-creation)
+  - [Workers Communication](#workers-communication)
+# Object Search Service
+## Motivation
+Object search is an important technology in the field of computer vision, which is widely used in security monitoring, intelligent transportation, etc. Generally, online object search applications have stringent latency constraints, which cannot be met by cloud computing schemes. Object search schemes based on edge computing have the characteristics of low latency and data privacy security, and are the mainstream technology trend. However, the amount of feature matching computation increases exponentially with the expansion of the search object scale, and a single edge computing node is difficult to support large-scale object search. Multi-edge collaborative computing can accelerate large-scale object search applications and improve the search accuracy, which is the future development trend. We propose the first open source end-to-end multi-edge collaborative object search solution. Based on KubeEdge's cloud-edge collaboration and resource management capabilities, we utilize multiple edge computing nodes to execute the AI inference tasks of object search in parallel. Our solution can not only reduce delay and improve throughput, but also will bring accuracy promotion. In addition, our solution will also support efficient offline object search.
+### Goals
+* Support single/multi-object search
+* Support across-camera object search
+* Support parallel ReID
+* Support historical video data object search
+* Support multi-camera data joint analysis and decision making
+* Provide a user entry for submitting search tasks and result queries
+
+
+## Proposal
+We propose using Kubernetes Custom Resource Definitions (CRDs) to describe the object search service specification/status and a controller to synchronize these updates between edge and cloud.
+![](./images/object-search-service-crd.png)
+
+### Use Cases
+
+* User can create typical multi-edge collaborative object search applications with providing AI models.
+
+## Design Details
+### CRD API Group and Version
+The `ObjectSearchService` CRD will be namespace-scoped.
+The tables below summarize the group, kind and API version details for the CRD.
+
+* ObjectSearchService
+
+| Field                 | Description             |
+|-----------------------|-------------------------|
+|Group                  | sedna.io     |
+|APIVersion             | v1alpha1                |
+|Kind                   | ObjectSearchService   |
+
+### Object search service CRD
+Below is the CustomResourceDefinition yaml for `ObjectSearchService`:
+
+[crd source](/build/crd-samples/sedna/objectsearchservice_v1alpha1.yaml)
```

Comment:
this crd source has the wrong link, should be generated by `make crds`, so I think we can omit this section.